starfish/imagestack/imagestack.py

```diff
@@ -177,26 +159,15 @@ def __init__(self, image_partition: TileSet) -> None:
             },
         )
 
-        # iterate through the tiles and set the data.
-        for tile in self._image_partition.tiles():
-            h = tile.indices[Indices.ROUND]
-            c = tile.indices[Indices.CH]
-            zlayer = tile.indices.get(Indices.Z, 0)
-            data = tile.numpy_array
-
-            if max_size != data.dtype.itemsize:
-                warnings.warn(
-                    f"Tile "
-                    f"(R: {tile.indices[Indices.ROUND]} C: {tile.indices[Indices.CH]} "
-                    f"Z: {tile.indices[Indices.Z]}) has "
-                    f"dtype {data.dtype}.  One or more tiles is of a larger dtype "
-                    f"{self._data.dtype}.",
-                    DataFormatWarning)
-
-            data = img_as_float32(data)
-            self.set_slice(indices={Indices.ROUND: h, Indices.CH: c, Indices.Z: zlayer}, data=data)
+        r_c_z_combinations = list(product(
```

Comment:
If I'm reading this right you could just use _iter_indices() over r,ch,z here. I think it does the same calculation.

Reply:
You are right.  Great idea.